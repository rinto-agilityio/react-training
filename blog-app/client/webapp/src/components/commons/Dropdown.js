import React, { memo, useRef, useEffect, useState } from 'react';
import PropTypes from 'prop-types';

// import style
import './styles/DropdownStyle.css'

export const Dropdown = props => {
  const { items, classWrapper, classItem } = props;

  const node = useRef();
  const [open, setOpen] = useState(false);

  useEffect(() => {
    // add when mounted
    document.addEventListener("mousedown", handleClick);
    // return function to be called when unmounted
    return () => {
      document.removeEventListener("mousedown", handleClick);
    };
  }, []);

  const handleClick = e => {
    if (!node.current.contains(e.target)) {
      setOpen(false)
    }
  }

  const handleClickDropdown = () => {
    setOpen(!open)
  }

  return (
    <div ref={node} className='dropdown-wrap'>
     <div onClick={handleClickDropdown} className='arrow-down'></div>
      {
        open &&
        <ul
          className={`dropdown-menu-wrap ${classWrapper ? ` ${classWrapper}` : ''}`}
        >
          {items.length > 0 &&
            items.map(item => (
              <li
                key={item}
                className={`dropdown-item${
                  classItem ? ` ${classItem}` : ''
                }`}
              >
                {item}
              </li>
            ))}
        </ul>
      }
    </div>
  );
};

Dropdown.propTypes = {
  items: PropTypes.array,
  classWrapper: PropTypes.string,
  classItem: PropTypes.string,
};

Dropdown.defaultProps = {
  items: [],
  classWrapper: null,
  classItem: null,
};

export default memo(Dropdown);
