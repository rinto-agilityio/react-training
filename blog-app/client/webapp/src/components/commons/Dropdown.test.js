//components
import Dropdown from './Dropdown';

describe('Dropdown Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Dropdown>Facebook</Dropdown>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

})
