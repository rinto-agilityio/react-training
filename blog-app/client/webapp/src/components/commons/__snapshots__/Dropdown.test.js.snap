// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component renders correctly 1`] = `
<div
  className="dropdown-wrap"
>
  <div
    className="arrow-down"
    onClick={[Function]}
  />
</div>
`;
